### ISSUE TITLE
Mammotion Luba

### ISSUE THREAD

**Issue by Bernd1984** [ioBroker/AdapterRequests#852]
Hi,
Ich würde es sehr begrüßen, wenn sich jemand finden würde, der den Mähroboter "Luba" von Mammotion in iOBroker einbinden könnte.
Der Mäher ist via wifi und vorallem Bluetooth mit der Smartphone App verbunden
Leider ist keine API verfügbar. Auf Github ist ein reverse Engineering Ansatz angedacht, aber dort tut sich nicht wirklich etwas.
https://github.com/02JanDal/pyluba

Den Mäher selbst kann ich euch schwer empfehlen. Lediglich die Smarthome Anbindung ist aktuell ein Manko - bis jetzt? ;-)
Grüße

**Comment by Kaiiide** [comment]
Würde ich auch sehr begrüßen 👯

**Comment by tiigaaa** [comment]
Hallo,
ich würde das auch sehr begrüßen. 
Nachdem das Teil ja etliche haben hoffe ich stark daß was kommen wird. ☺️

**Comment by mcm1957** [comment]
Instead of me too posts please click at thumbs up at original request post to show your interrest.

**Comment by TomasiDeveloping** [comment]
Wäre eine gute Sache. Habe den Luba erst paar Tage. Über die API hab ich das Login hinbekommen aber mehr nicht wirklich da viele POST Requests sind und eine Body brauche der glaub nicht bekannt ist.
Das meiste geht aber auch über Bluetooth und das ist nicht so Ideal zum Integrieren.
Auch beim Luba Support selbst auf die Frage zu einer API Doku kahm ein nein zurück.

**Comment by FredoFuchs** [comment]
wie ist das denn mit Bluetooth falls luba sich auch in Zukunft nicht per WLAN steuern lässt ? Ich nutze zu Hause homeassistant und habe ein esp-Board als bt-gateway zur Abfrage von Bluetooth Sensoren. Also das gateway leitet quasi die Bluetooth infos weiter zum Smarthome-server.
Könnte man sowas auch nutzen im z.B. anstatt mit dem Handy dann damit die bluetooth verbindung zum luba aufzubauen und die Steuerung über Homeassistant oder iobroker umsetzen ?

**Comment by Bernd1984** [comment]
Ja, Fredo, das wäre genau der Plan, nur ist das Reverse Engineering von Bluetooth Befehlen scheinbar ziemlich schwer. Schade, dass Mammotion keine API zur Verfügung stellt um die Mäher per WiFi oder Bluetooth steuern zu können. Ich hoffe, es findet sich ein kluger Kopf, der hier was zaubern kann. Ich kanns leider nicht.

**Comment by Bernd1984** [comment]
Jemand ist schon relativ weit mit dem reverse engineering. 

https://github.com/mikey0000/Luba-API

**Comment by tiigaaa** [comment]
Hi Leute,
gibt e szu dem Thema schon ein update?
Der Cuder vom Repo scheint ja noch voll daradn zu arbeiten aber da steht so weinig Info drin das ich zumindest nicht verstehe wie man das anwenden könnte.

Alexa soll ja noch nicht gehen, und somit weis ja auch keiner was dann via Alexa kommt, MQTT wär da schon viel besser denk ich.

Also danke schonmal, falls sich was ergibt lasst es mich wissen. :) Dann kauf ich das Ding


LG

**Comment by mcm1957** [comment]
Also HIER hat niemand sich gemeldet und bekannt gegeben dass er / sie dran arbeitet.
'Cuder vom Repo' sagt mir mal nichts.
Falls jamand an einem Adapter arbeitet, bitte hie rmelden damit das Issue zugeteilt und er Status geschaltet werden kann. Erspart potenzielle Doppelgleisigkeiten.

**Comment by Seide-Th** [comment]
Hallo, ich würde mich auch freuen wenn der Luba2 im iobroker einzubinden ist. Meiner ist heute gekommen und ich werde ihn dann nächste Woche mal in Betrieb nehmen.

**Comment by mibr85** [comment]
check out it getting forward
https://github.com/mikey0000/Mammotion-HA

**Comment by Gukerl** [comment]
YUKA Mini 600 ist der Meinige. Ich denke, da wird nicht viel Unterschied sein.

**Comment by mcm1957** [comment]
Ok - keine Ahnung auf welchen Coder sich das bezogen hat.

Fact ist jedenfalls dass mir nicht bekannt wäre dass jemand an einem 'Luba' Adapter für ioBroker arbeitet.

Und bitte statt 'Me too posts das Thumbs up Icon im ERSTEN Beitrag klicken.

**Comment by mcm1957** [comment]
Repo: https://github.com/DNAngelX/ioBroker.mammotion

Review / PR for review and adding to repository still missing
